x/expiration/keeper/keeper.go

```diff
@@ -0,0 +1,458 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	authzkeeper "github.com/cosmos/cosmos-sdk/x/authz/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/provenance-io/provenance/x/expiration/types"
+	metadatatypes "github.com/provenance-io/provenance/x/metadata/types"
+)
+
+// Handler is a name record handler function for use with IterateExpirations.
+type Handler func(record types.Expiration) error
+
+// Keeper defines the name module Keeper
+type Keeper struct {
+	// The reference to the Paramstore to get and set account specific params
+	paramSpace paramtypes.Subspace
+
+	// Key to access the key-value store from sdk.Context.
+	storeKey storetypes.StoreKey
+
+	// The codec for binary encoding/decoding.
+	cdc codec.BinaryCodec
+
+	// To check granter grantee authorization of messages
+	authzKeeper authzkeeper.Keeper
+
+	// To handle account interactions.
+	acctKeeper banktypes.AccountKeeper
+
+	// To handle expiration deposit processing.
+	bankKeeper bankkeeper.Keeper
+
+	// Message service router
+	router baseapp.IMsgServiceRouter
+}
+
+// NewKeeper returns an expiration keeper. It handles:
+// - managing a hierarchy of expiration
+// - enforcing permissions for expiration creation/extension/deletion
+//
+// CONTRACT: the parameter Subspace must have the param key table already initialized
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	key storetypes.StoreKey,
+	paramSpace paramtypes.Subspace,
+	authzKeeper authzkeeper.Keeper,
+	acctKeeper banktypes.AccountKeeper,
+	bankKeeper bankkeeper.Keeper,
+	router baseapp.IMsgServiceRouter,
+) Keeper {
+	// set KeyTable if it has not already been set
+	if !paramSpace.HasKeyTable() {
+		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	// ensure governance module account is set
+	if addr := acctKeeper.GetModuleAddress(types.ModuleName); addr == nil {
```

Comment:
It doesn't look like addr is being used. Can you just change this to `if acct.Keeper.GetModuleAddress(types.ModuleName) == nil`